CQL binary parser ignores Tracing flag in the response if request didn't have the Tracing flag set

*Setup*
gocql tree: 
 * HEAD: 2faf01b9681e19c25b01de3b7e78f90447d71869

go application tree: 
 * URL: https://github.com/tomer-sandler/scylla-code-samples
 * HEAD: 7dbd4fef76926372457acfe7794d3e829ea4b075

scylla tree:
 * HEAD: 85a127b

*Description*
When testing the Go CQL driver with scylla (https://github.com/scylladb/scylla) the application crashed when we enabled the probabilistic Tracing or Slow Query Log (e.g. `nodetool settraceprobability 0.001`):

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x493e11]

goroutine 11 [running]:
panic(0x6edd40, 0xc420012140)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/gocql/gocql.(*Conn).prepareStatement(0xc420196000, 0x0, 0x0, 0x74d007, 0x5e, 0x0, 0x0, 0xc42008fba0, 0x410a49, 0xc4200dc538)
        /home/vladz/projects/src/src/github.com/gocql/gocql/conn.go:728 +0x831
github.com/gocql/gocql.(*Conn).executeQuery(0xc420196000, 0xc420162200, 0xc420196000)
        /home/vladz/projects/src/src/github.com/gocql/gocql/conn.go:783 +0x149
github.com/gocql/gocql.(*controlConn).query.func1(0xc420196000, 0x72de60)
        /home/vladz/projects/src/src/github.com/gocql/gocql/control.go:409 +0x34
github.com/gocql/gocql.(*controlConn).withConn(0xc420132480, 0xc4200fc490, 0x5e)
        /home/vladz/projects/src/src/github.com/gocql/gocql/control.go:397 +0x12b
github.com/gocql/gocql.(*controlConn).query(0xc420132480, 0x74d007, 0x5e, 0x0, 0x0, 0x0, 0x569125)
        /home/vladz/projects/src/src/github.com/gocql/gocql/control.go:410 +0x132
github.com/gocql/gocql.(*controlConn).fetchHostInfo.func1(0xc4201580a0, 0xc4201580a0, 0x10)
        /home/vladz/projects/src/src/github.com/gocql/gocql/control.go:440 +0x5e
github.com/gocql/gocql.(*controlConn).fetchHostInfo(0xc420132480, 0xc4200fc460, 0x10, 0x10, 0x2352, 0x1, 0xc4200fc447, 0x1)
        /home/vladz/projects/src/src/github.com/gocql/gocql/control.go:457 +0x161
github.com/gocql/gocql.(*Session).handleNewNode(0xc42001f8c0, 0xc4200fc460, 0x10, 0x10, 0x2352, 0x2300)
        /home/vladz/projects/src/src/github.com/gocql/gocql/events.go:180 +0x3b8
github.com/gocql/gocql.(*Session).handleNodeUp(0xc42001f8c0, 0xc4200fc460, 0x10, 0x10, 0x2352, 0x0)
        /home/vladz/projects/src/src/github.com/gocql/gocql/events.go:262 +0x243
github.com/gocql/gocql.(*controlConn).setupConn(0xc420132480, 0xc420196000, 0x1, 0x1)
        /home/vladz/projects/src/src/github.com/gocql/gocql/control.go:261 +0x179
github.com/gocql/gocql.(*controlConn).connect(0xc420132480, 0xc4200341c0, 0x1, 0x1, 0x4, 0x0)
        /home/vladz/projects/src/src/github.com/gocql/gocql/control.go:231 +0x19a
github.com/gocql/gocql.(*Session).init(0xc42001f8c0, 0xc4201520c0, 0x0)
        /home/vladz/projects/src/src/github.com/gocql/gocql/session.go:173 +0x3b3
github.com/gocql/gocql.NewSession(0xc4200cbdd0, 0x1, 0x1, 0x73c8d9, 0x5, 0x0, 0x37e11d600, 0x23c34600, 0x2352, 0x0, ...)
        /home/vladz/projects/src/src/github.com/gocql/gocql/session.go:137 +0x871
github.com/gocql/gocql.(*ClusterConfig).CreateSession(0xc420092480, 0x767ff0, 0xc4200cbe80, 0x0)
        /home/vladz/projects/src/src/github.com/gocql/gocql/cluster.go:153 +0x62
main.main.func3(0xc4200cbe80, 0xc420092480, 0xc4200cbe90, 0xc42000c600, 0xc42000c660, 0x0, 0xc420013288)
        /home/vladz/work/tomer/scylla-code-samples/efficient_full_table_scan_example_code/efficient_full_table_scan.go:217 +0x99
created by main.main
        /home/vladz/work/tomer/scylla-code-samples/efficient_full_table_scan_example_code/efficient_full_table_scan.go:254 +0x841
```

This was caused by the fact that scylla was putting the Tracing session ID in the response when the probabilistic tracing was enabled. 

Although this scylla behaviour is suboptimal (and we are going to fix that: https://github.com/scylladb/scylla/issues/2179) this is absolutely legal and this definitely should not cause any crashes.
The CQL binary protocol parser should obey the values of the `flags` field in the header according to the chapter 2.2 in the CQL protocol specification (https://github.com/apache/cassandra/blob/trunk/doc/native_protocol_v4.spec):
```
If a response frame has the tracing flag set, its body contains a tracing ID. The tracing ID is a [uuid] and is the first thing in
the frame body. The rest of the body will then be the usual body corresponding to the response opcode.
```

--- a/conn.go
+++ b/conn.go
@@ -724,7 +724,7 @@ func (c *Conn) prepareStatement(ctx context.Context, stmt string, tracer Tracer)
 
 	// TODO(zariel): tidy this up, simplify handling of frame parsing so its not duplicated
 	// everytime we need to parse a frame.
-	if len(framer.traceID) > 0 {
+	if len(framer.traceID) > 0 && tracer != nil {
 		tracer.Trace(framer.traceID)
 	}
 

